components/support/jwcrypto/src/aes.rs

```diff
@@ -0,0 +1,84 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// Wrappers we use around rc_crypto's AES implementation. Specifically,
+// "enc=A256GCM" from RFC7518, Section 4.7 - for all the gory details, see
+// https://tools.ietf.org/html/rfc7518#section-4.7.
+
+use crate::{
+    error::{JwCryptoError, Result},
+    CompactJwe, EncryptionAlgorithm, JweHeader,
+};
+use rc_crypto::{aead, rand};
+
+/// Does the AES-encrypt heavy-lifting for the schemes supported by this crate.
+pub(crate) fn aes_gcm_encrypt(
+    data: &[u8],
+    protected_header: JweHeader,
+    secret: &[u8],
```

Comment:
nit: here and below, if we rename `secret` to `content_encryption_key` then I think it will be slightly easier for future readers to mentally map between the spec and the code here.